website/static/js/passwordForms.js

```diff
@@ -119,6 +119,30 @@ var ChangePasswordViewModel = oop.extend(BaseViewModel, {
 var SetPasswordViewModel = oop.extend(BaseViewModel, {
     constructor: function () {
         var self = this;
+        // Call constructor at the beginning so that self.password exists
+        self.super.constructor.call(this);
```

Comment:
Are you certain that this doesn't break validation? Other ViewModels have a comment that indicate that the constructor needs to be called after the observables are declared: https://github.com/CenterForOpenScience/osf.io/blob/607e4d7ee05b4e310a7400b0183c8a68dcc7e92c/website/static/js/passwordForms.js#L107-L108